client/state/index.js

```diff
@@ -132,5 +143,75 @@ export function createReduxStore( initialState = {} ) {
 		isBrowser && window.devToolsExtension && window.devToolsExtension()
 	].filter( Boolean );
 
-	return compose( ...enhancers )( createStore )( reducer, initialState );
+	return enhancers;
 }
+
+/**
+ * Named export function for creating a redux store.
+ * This creates a new redux store within the calypsoStore instance.
+ *
+ * @param { object } initialState Redux state with which to initialize (optional).
+ * @param { array } startingReducers Initial list of reducers to start with.
+ * @param { array } enhancers Assembled list of enhancers for the store.
+ * @return { object } The redux store that was created.
+ */
+export function createReduxStore( initialState, startingReducers = initialReducers, enhancers = createEnhancers() ) {
+	// This creates a normal redux store, but it adds our list of original reducers.
+	// This is so we can modify that list of reducers later on and regenerate it.
+	const reduxStore = compose( ...enhancers )( createStore )( combineReducers( startingReducers ), initialState );
+	reduxStore.reducers = startingReducers;
+
+	/**
+	 * Adds a reducer dynamically to the store.
+	 *
+	 * @param { string } name The top-level context name for the reducer.
+	 * @param { function } func The reducer function, in the format of ( state, action ), returning state.
+	 */
+	function addReducer( name, func ) {
+		if ( reduxStore.reducers[ name ] !== undefined ) {
+			throw new Error( `addReducer(): Reducer by name "${ name }" already in use` );
+		}
```

Comment:
I thought about this before, but it really does open a new can of worms. If we have two pieces of code call `addReducer`, presumably they might also call `removeReducer`. What happens when code A yanks the reducer out of the chain and code B still needs it?

In order for it all to work this way, we will have to disallow reducers to be removed from the chain, and do we really want that? I feel it will catch up with us eventually and we're already pushing the limits of memory and time delays.